### PR TITLE
Use shared libs

### PR DESCRIPTION
**Pull request by daranday**

### CODE REVIEW

**Review by FrankieLi — COMMENTED** [review]
Umm... it says all checks failed? Should it build in Jenkins?

**Comment by daranday** [comment]
@FrankieLi Yep I'm trying to debug.

**Review by FrankieLi — APPROVED** [review]
Talked in person. Looks good to me.

File: Jenkinsfile [hunk 1]
```diff
@@ -2,145 +2,106 @@
...
-        def repo = git_info.is_release ? build_config.repo : build_config.staging_repo
-
-        withAWS(credentials:'package-uploads') {
-            sh("./publish.sh ${repo} ${build_config.dist}")
-        }
-    }
+stage("Tag and deploy?") {
+  deploy_mode = "SKIP"
+  if (git_info.is_release_branch) {
+    ditto_utils.checkReleaseBranch(git_info.branch, version)
```
Comment (oselivanov) [inline Jenkinsfile]: This is broken, no such function.

**Review by oselivanov — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (daranday) [reply Jenkinsfile]: Fixed!

File: Jenkinsfile [hunk 2]
```diff
@@ -2,145 +2,106 @@
...
+
+        image_name =
+          ditto_utils.buildDockerImageName(git_info.repo_name, platform)
+        ditto_deb.buildInsideDocker(image_name, build_config.docker_file)
+        ditto_deb.generatePackageInsideDocker(image_name, version, revision)
+        ditto_deb.publishPackageToS3(build_config.apt_test_repo,
+                                     build_config.dist)
+      }
+
+      stage("Install from ${platform} repo and test") {
```
Comment (oselivanov) [inline Jenkinsfile]: Installing

File: Jenkinsfile [hunk 3]
```diff
@@ -2,145 +2,106 @@
...
+  }
 }
 
-def cleanUp(target) {
-    stage("CleanUp ${target}") {
-        def current_dir = pwd()
-        echo "Cleaning up ${current_dir}"
-        deleteDir()
+node('build && docker') {
+  stage("Build and Publish to Test") {
```
Comment (oselivanov) [inline Jenkinsfile]: Building and publishing

File: Jenkinsfile [hunk 4]
```diff
@@ -2,145 +2,106 @@
...
-
-    stage("ArchiveArtifacts ${target}") {
-        archiveArtifacts(artifacts: 'build/*.deb')
+  BUILD_CONFIGS.each { platform, build_config ->
+    dir(platform) {
+      stage("Checking out ${platform}") {
+        git_info = ditto_git.checkoutRepo()
+      }
+
+      stage("Building and publishing dev revision ${platform}") {
```
Comment (oselivanov) [inline Jenkinsfile]: Building and publishing ${platform} dev revision

File: Jenkinsfile [hunk 5]
```diff
@@ -2,145 +2,106 @@
...
+      }
     }
-    return tag
-}
+    ditto_git.pushTag(tag)
+  }
 
-def getGitBranch() {
-    return sh(script: "git rev-parse --abbrev-ref HEAD", returnStdout: true).trim()
+  stage("Clean up") {
```
Comment (oselivanov) [inline Jenkinsfile]: Cleaning up

**Review by oselivanov — APPROVED** [review]
Checked, Jenkinsfile part, lgtm

File: Jenkinsfile (hunk 4 above)
Comment (daranday) [reply Jenkinsfile]: Done

File: Jenkinsfile (hunk 2 above)
Comment (daranday) [reply Jenkinsfile]: done

File: Jenkinsfile (hunk 3 above)
Comment (daranday) [reply Jenkinsfile]: done

File: Jenkinsfile (hunk 5 above)
Comment (daranday) [reply Jenkinsfile]: done

**Comment by oselivanov** [comment]
Ship it!

**Pull request merged**